libs/math/functions.h

```diff
@@ -152,6 +158,42 @@ sinc (T const& x)
     return (x == T(0) ? T(1) : std::sin(x) / x);
 }
 
+/* ----------------------------- Popcount ------------------------- */
+
+/** Returns the number of one bits of an integer. */
+template <typename T>
+std::size_t constexpr
+popcount (T const x)
+{
+    static_assert(std::is_integral<T>::value && std::is_unsigned<T>::value,
+        "T must be an unsigned integral type.");
```

Comment:
Added requirement for T to be unsigned.
Without this, MSVS would silently use the unoptimized bitset::count for all signed integers.

I tried adding overloading for the signed types on MSVS, but this increased the line count a bit too much for little added benefit.